framework/src/main/java/org/checkerframework/framework/util/QualifierHierarchyMostlyWithoutElements.java

```diff
@@ -0,0 +1,149 @@
+package org.checkerframework.framework.util;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.util.Elements;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.framework.qual.AnnotatedFor;
+import org.checkerframework.framework.type.QualifierHierarchy;
+
+/**
+ * A {@link QualifierHierarchy} where qualifiers may be represented by annotations with elements,
+ * but most of the qualifiers do not have elements. In contrast to {@link
```

Comment:
I don't really have any more specific advice.  When I was convert all of the MultiGraphHierarchy subclass to one of the new classes, I only used this class when all but 1 or 2 qualifiers have elements.  Otherwise, it was easier to use QualifierHierarchyWithElements and write out all cases.